site/src/components/pages/user/useUserStatus.ts

```diff
@@ -0,0 +1,13 @@
+import { useUser } from "../../../context/UserContext";
+import { SerializedUser } from "../../../lib/api/model/user.model";
+
+export const useUserStatus = (
+  user: SerializedUser,
+): "loading" | "current" | "other" => {
```

Comment:
Having three statuses instead of two helps avoid UI flickering for logged in users. If `useUser()` is still loading and this hook returns ’it’s not us’, placeholder content would change shortly after loading.